lib/assets/javascripts/_modules/search.js

```diff
@@ -0,0 +1,171 @@
+//= require lunr.min
+//= require _vendor/jquery.mark.js
+(function($, Modules) {
+  'use strict';
+
+  Modules.Search = function Search() {
+    var $html = $('html');
+    var lunrIndex;
+    var lunrData;
+    var $searchForm;
+    var $searchInput;
+    var $searchResults;
+    var $searchResultsTitle;
+    var $searchResultsWrapper;
+    var $searchResultsClose;
+    var results;
+    var maxSearchEntries = 10;
+
+    this.start = function start($element) {
+      $searchForm = $element.find('form');
+      $searchInput = $element.find('#search');
+      $searchResultsWrapper = $element.find('.search-results')
+      $searchResults = $searchResultsWrapper.find('.search-results__content');
+      $searchResultsTitle = $searchResultsWrapper.find('.search-results__title');
+      $searchResultsClose = $searchResultsWrapper.find('.search-results__close');
+      downloadSearchIndex();
+      attach();
+    };
+
+    function downloadSearchIndex() {
+      updateTitle('Loading search index')
+      $.ajax({
+        url: '/search.json',
+        cache: true,
+        method: 'GET',
+        success: function(data) {
+          lunrData = data;
+          lunrIndex = lunr.Index.load(lunrData.index);
+          $(document).trigger('lunrIndexLoaded');
+        }
+      });
+    }
+
+    function attach() {
+      // Search functionality on search text input
+      $searchInput.on('input', function (e) {
+        e.preventDefault();
+        var query = $(this).val();
+        search(query);
+      });
+
+      // Set focus on the first search result instead of submiting the search  
+      // form to Google
+      $searchForm.on('submit', function(e) {
+        e.preventDefault();
+        $searchResults.find('.search-result__title a').first().focus();
+      });
+
+      // Closing the search results, move focus back to the search input
+      $searchResultsClose.on('click', function(e) {
+        e.preventDefault();
+        $searchInput.focus();
+        hideResults();
+      });
+    }
+
+    function searchIndex(query) {
```

Comment:
I think `searchIndex` is a weird name for this function, since it doesn't return the search index, or do any indexing, it just does a search. Maybe `getResults` would be clearer.